File: frontend/src/types/quiz.ts
// Quiz data interfaces for the CodingAgent module

export type QuizSubtype = 'MCQ' | 'TrueFalse' | 'FillBlanks' | 'MatchFollowing';

export interface Theme {
  primaryColor: string;
  secondaryColor: string;
  backgroundColor: string;
  textColor: string;
  fontFamily: string;
  animation?: string;
  iconSet?: string;
}

export interface MCQQuestion {
  id: number;
  question: string;
  options: string[];
  correctAnswer: number; // index of correct option
  explanation?: string;
}

export interface TrueFalseQuestion {
  id: number;
  statement: string;
  correctAnswer: boolean;
  hint?: string;
  explanation?: string;
}

export interface FillBlanksQuestion {
  id: number;
  sentence: string; // with blanks marked as [BLANK]
  correctAnswers: string[];
  hint?: string;
  explanation?: string;
}

export interface MatchPair {
  id: number;
  left: string;
  right: string;
}

export interface MatchFollowingQuestion {
  id: number;
  instruction: string;
  pairs: MatchPair[];
}

export interface QuizData {
  subtype: QuizSubtype;
  theme: Theme;
  title: string;
  description?: string;
  questions: MCQQuestion[] | TrueFalseQuestion[] | FillBlanksQuestion[] | MatchFollowingQuestion[];
  timeLimit?: number; // in minutes
  passingScore?: number; // percentage
}

export interface QuizComponentProps {
  quizData: QuizData;
  isOpen: boolean;
  onClose: () => void;
  onComplete?: (score: number, totalQuestions: number) => void;
} 

File: backend/prompts/quiz.py
COMPREHENSIVE_QUIZ_PROMPT = """You are an expert educational content creator for an educational AI platform. Analyze the provided content and create a quiz with EXACTLY 10 questions in the most suitable format.

## EXACT JSON FORMAT REQUIRED:

### MCQ Format:
```json
{
  "subtype": "MCQ",
  "theme": {
    "primaryColor": "#16213e",
    "secondaryColor": "#0f3460", 
    "backgroundColor": "#1a1a2e",
    "textColor": "#e0dede",
    "fontFamily": "Arial",
    "animation": "slide-in"
  },
  "title": "[Title based on content]",
  "description": "[Description based on content]",
  "questions": [
    {
      "id": 1,
      "question": "[Question from content]",
      "options": ["[Option 1]", "[Option 2]", "[Option 3]", "[Option 4]"],
      "correctAnswer": 1,
      "hint": "[Helpful hint from content]",
      "explanation": "[Explanation from content]"
    }
  ]
}
```

### TrueFalse Format:
```json
{
  "subtype": "TrueFalse",
  "theme": {
    "primaryColor": "#16213e",
    "secondaryColor": "#0f3460", 
    "backgroundColor": "#0f0f0f",
    "textColor": "#d0f0ff",
    "fontFamily": "Arial",
    "animation": "slide-in"
  },
  "title": "[Title based on content]",
  "description": "[Description based on content]",
  "questions": [
    {
      "id": 1,
      "statement": "[Statement from content]",
      "correctAnswer": true,
      "hint": "[Hint from content]",
      "explanation": "[Explanation from content]"
    }
  ]
}
```

### FillBlanks Format:
```json
{
  "subtype": "FillBlanks",
  "theme": {
    "primaryColor": "#16213e",
    "secondaryColor": "#0f3460", 
    "backgroundColor": "#1c1c1c",
    "textColor": "#f0e6d2",
    "fontFamily": "Arial",
    "animation": "slide-in"
  },
  "title": "[Title based on content]",
  "description": "[Description based on content]",
  "questions": [
    {
      "id": 1,
      "sentence": "[Sentence with [BLANK] from content]",
      "correctAnswers": ["[Answer from content]"],
      "hint": "[Hint from content]",
      "explanation": "[Explanation from content]"
    }
  ]
}
```

### MatchFollowing Format:
```json
{
  "subtype": "MatchFollowing",
  "theme": {
    "primaryColor": "#16213e",
    "secondaryColor": "#0f3460", 
    "backgroundColor": "#202020",
    "textColor": "#ffffff",
    "fontFamily": "Arial",
    "animation": "slide-in"
  },
  "title": "[Title based on content]",
  "description": "[Description based on content]",
  "questions": [
    {
      "id": 1,
      "instruction": "[Instruction from content]",
      "pairs": [
        { "id": 1, "left": "[Item from content]", "right": "[Match from content]" },
        { "id": 2, "left": "[Item from content]", "right": "[Match from content]" },
        { "id": 3, "left": "[Item from content]", "right": "[Match from content]" },
        { "id": 4, "left": "[Item from content]", "right": "[Match from content]" },
        { "id": 5, "left": "[Item from content]", "right": "[Match from content]" }
      ]
    }
  ]
}
```

## REQUIREMENTS:
1. **EXACTLY 10 questions** (ids 1-10)
2. Choose the best format based on content
3. Generate questions from the actual content provided
4. Use the exact JSON structure shown above
5. Include the "animation": "slide-in" field in theme
6. **IMPORTANT: Choose appropriate colors based on content:**
   - Science/Technology: Purple/Blue themes (#9c27b0, #7b1fa2, #4a148c)
   - History/Literature: Warm themes (#ff5722, #e64100, #bf360c)
   - Education/Learning: Blue themes (#2196f3, #1976d2, #0d47a1)
   - General Knowledge: Green themes (#4caf50, #388e3c, #1b5e20)
   - **AI should analyze content and choose the most fitting color scheme**
7. **IMPORTANT: Create helpful, specific hints** - NOT generic ones like "think logically"
   - For TrueFalse: Give contextual clues from the content (e.g., "Consider the historical period mentioned" or "Think about the scientific property described")
   - For FillBlanks: Give word association or definition hints (e.g., "This word rhymes with..." or "This is a type of...")
   - For MCQ: Give process of elimination hints or conceptual guidance
   - Hints should guide without revealing the answer directly

## OUTPUT:
Return ONLY the JSON. Do not add any text before or after.
""" 